q2_metadata/normalization/_norm_check_rules.py

```diff
@@ -0,0 +1,311 @@
+        'format': 'allowed'
+    }
+
+    if rule not in rule_type_map:
+        raise RuleError(rule, variable, rule_value)
+
+    if rule in ['blank', 'missing']:
+        rule_method_key = 'check_blank_missing'
+    else:
+        rule_method_key = 'check_%s' % rule
+
+    rule_method = globals()[rule_method_key]
+    rule_error = globals()['%sError' % rule.capitalize()]
+    rule_type = rule_type_map[rule]
+    if rule_method(rule_value):
+        rules[rule_type][rule] = rule_value
+    else:
+        raise rule_error(variable, rule_value)
+
+
+def check_expected(rule_value):
+    """Check that the user-defined value for the
+    rule "expected" is correctly formatted.
+
+    expected must be a list
+        Apply rule of allowing only expected values:
+        These do allow the missing and blanks.
+                    expected:
+                        - str
+                        - list
+
+    Parameters
+    ----------
+    rule_value
+        Parsed rule value for the rule "expected".
+
+    Returns
+    -------
+    bool
+        True if successful, False otherwise.
+    """
+    # check if the rule value is indeed a list (it must be the case).
+    if isinstance(rule_value, list):
+        # checks that all the items in the expected value list are strings.
+        str_items = [x for x in rule_value if isinstance(x, str)]
+        if len(str_items) == len(rule_value):
+            return True
+    return False
+
+
+def check_ontology(rule_value):
+    """Check that the user-defined value for the
+    rule "ontology" is correctly formatted.
+
+    ontology must be a str
+        Name of the ontology to lookup within for control vocabulary, e.g.
+                    ontology: Gazetteer
+
+    Parameters
+    ----------
+    rule_value
+        Parsed rule value for the rule "ontology".
+
+    Returns
+    -------
+    bool
+        True if successful, False otherwise.
+    """
+    if isinstance(rule_value, str):
+        if rule_value in ['Gazetteer ontology']:
+            return True
+    return False
+
+
+def check_remap(rule_value):
+    """Check that the user-defined value for the
+    rule "remap" is correctly formatted.
+
+    remap must be a dict
+        Apply the rule of remapping variables values: replace
+        all instances of a given value by another one value.
+                    remap:
+                      str: str
+
+    Parameters
+    ----------
+    rule_value
+        Parsed rule value for the rule "remap".
+
+    Returns
+    -------
+    bool
+        True if successful, False otherwise.
+    """
+    if isinstance(rule_value, dict):
+        str_values = [
+            x for x, y in rule_value.items() if
+                isinstance(x, (str, int, float)) and
+                isinstance(y, (str, int, float))
+        ]
+        if len(str_values) == len(rule_value.keys()):
+            return True
+    return False
+
+
+def check_validation(rule_value):
+    """Check that the user-defined value for the
```

Comment:
Make sure to reformat this docstring to follow some of the suggestions above.

Reply:
done:
```
    """Check that the user-defined value for the rule 
    "validation" is a nested dictionary that has for first 
    key "force_to_blank_if" and for nested key "is null", 
    and for nested value a list. 

    Parameters
    ----------
    rule_value
        Parsed rule value for the rule "validation".

    Returns
    -------
    bool
        True if successful, False otherwise.
    """
```